ui/src/app/components/workflows/workflows.component.ts

```diff
@@ -45,6 +47,27 @@ export class WorkflowsComponent implements AfterViewInit, OnDestroy {
     });
   }
 
+  isWorkflowHighlighted(id: number): boolean {
+    return this.router.url.split('/').some((part) => part === id.toString());
+  }
+
+  isProjectClosed(project: string, workflows: WorkflowIdentityModel[]): boolean {
+    const isOpen =
+      this.openedProjects.has(project) || workflows.some((workflow: WorkflowIdentityModel) => this.isWorkflowHighlighted(workflow.id));
+    if (isOpen && !this.openedProjects.has(project)) {
```

Comment:
Little bit redundant logic. If you expand this, it reads
`(A || B) && !A`, which is equivalent to `(A && !A) || (B && !A)`, which is equivalent to `B && !A`,
where `A = this.openedProjects.has(project)`, `B = workflows.some((workflow: WorkflowIdentityModel) => this.isWorkflowHighlighted(workflow.id))`.

So I would rewrite this if-condition like this:
`if(!this.openedProjects.has(project) && workflows.some((workflow: WorkflowIdentityModel) => this.isWorkflowHighlighted(workflow.id))) {`

Reply:
Nice, thanks. Fixed.